pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
i'm torn.  it is an input that describes the source being built, so in that sense it does belong here.  it is very similar to ContextDir in that sense.  it's also possible we might introduce new build types in the future that would also consume this value (for that matter CustomBuild could also want to consume this value).

However it does bother me having it as a first class field in this struct, given how specialized it is.

either way it's going to have to pass @smarterclayton 's api review, so @smarterclayton what's your view?